library/src/scala/quoted/Expr.scala

```diff
@@ -10,12 +10,34 @@ abstract class Expr[+T] private[scala] ()
 object Expr {
 
   /** `e.betaReduce` returns an expression that is functionally equivalent to `e`,
-   *   however if `e` is of the form `((y1, ..., yn) => e2)(e1, ..., en)`
-   *   then it optimizes this the top most call by returning the result of beta-reducing the application.
-   *   Otherwise returns `expr`.
+   *  however if `e` is of the form `((y1, ..., yn) => e2)(e1, ..., en)`
+   *  then it optimizes the top most call by returning the result of beta-reducing the application.
+   *  Similarly, all outermost curried function applications will be beta-reduced, if possible.
+   *  Otherwise returns `expr`.
    *
-   *   To retain semantics the argument `ei` is bound as `val yi = ei` and by-name arguments to `def yi = ei`.
-   *   Some bindings may be elided as an early optimization.
+   *  To retain semantics the argument `ei` is bound as `val yi = ei` and by-name arguments to `def yi = ei`.
+   *  Some bindings may be elided as an early optimization.
+   *
+   *  Example:
```

Comment:
I  still belive this is to complex for a first-time user. I would call this the general rule.
```suggestion
   *  Generally:
```

I would add a trivial beta reduction example before this.
````
Example:
```scala sc:nocompile
((a: Int, b: Int) => a + b).apply(x, y)
```
will be reduced to
```scala sc:nocompile
val a = x
val b = y
a + b
```
````

Reply:
Oh, I see that you did that in the other file. We should have the same example in both documentations. I believe that my new example will be better for new users that are less experienced.